### PR TITLE
chore: refactor to remove GasBlockStore

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- Replace `GasBlockStore` with `TrackingBlockStore`

**Reference issue to close (if applicable)**

Closes https://github.com/ChainSafe/forest/issues/2034

**Other information and links**

### CODE REVIEW

**Comment by lemmih** [comment]
Let's get rid of the `BlockStore` trait. It doesn't seem to be useful anymore.

**Comment by hanabi1224** [comment]
> Let's get rid of the BlockStore trait. It doesn't seem to be useful anymore.

Sure, would u suggest doing that in the same PR or a separate one?

**Comment by hanabi1224** [comment]
Also I think `ForestKernel` is not needed either, it's not instantiated anywhere in either forest for fvm code

File: vm/interpreter/src/fvm/externs.rs [hunk 1]
```diff
@@ -231,3 +222,89 @@ impl<DB: BlockStore> Consensus for ForestExterns<DB> {
...
+        write_bytes: &[usize],
+    ) -> anyhow::Result<()> {
+        let network_version = NetworkVersion::V8;
+        let stats = BSStats {
+            r: read_count,
+            w: write_bytes.len(),
+            br: 0, // Not used in current logic
+            bw: write_bytes.iter().sum(),
+        };
+        println!("stats: {:?}", stats);
```
Comment (LesnyRumcajs) [inline vm/interpreter/src/fvm/externs.rs:291]: do we need this print? I understand it's for debugging purposes only?

File: vm/interpreter/src/fvm/externs.rs [hunk 2]
```diff
@@ -231,3 +222,89 @@ impl<DB: BlockStore> Consensus for ForestExterns<DB> {
         Ok((cf, total_gas))
     }
 }
+
+fn cal_gas_used_from_stats(
```
Comment (LesnyRumcajs) [inline vm/interpreter/src/fvm/externs.rs:226]: It'd be nice to add some piece of documentation here. I don't get from a quick glimpse what is `r` and `w`, other than poorly named fields. We can't change them, but maybe we can at least explain stuff for those less bright like myself. :)

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: vm/interpreter/src/fvm/externs.rs (hunk 1 above)
Comment (hanabi1224) [reply vm/interpreter/src/fvm/externs.rs]: fixed

File: vm/interpreter/src/fvm/externs.rs (hunk 2 above)
Comment (hanabi1224) [reply vm/interpreter/src/fvm/externs.rs:226]: fixed

**Review by LesnyRumcajs — APPROVED** [review]
LGTM. I'm okay with addressing removing `BlockStore` trait either here or in a separate PR.

**Review by lemmih — APPROVED** [review]
Looking good. We can remove the unused trait and structs in a separate PR.

**Pull request merged**